### ISSUE TITLE
Argument list too long

### ISSUE THREAD

**Issue by manthesepeasrock** [JLLeitschuh/ktlint-gradle#326]
I have the pre-commit hook configured, and if for some reason a big change is made on the project the pre-commit script will pass every file changed as parameter, in my case I have a commit with 3266 staged files (ktlint added an end line to each and every one of our files) and this will make it impossible to make my commit

I went looking and the pre-commit script uses the following command `./gradlew --quiet ktlintFormat -PinternalKtlintGitFilter="$CHANGED_FILES"` and I can see the reason this does not use the directory instead on the files

Can this be changed so that the CHANGED_FILES values are iterated and `./gradlew --quiet ktlintFormat -PinternalKtlintGitFilter="$CHANGED_FILES_ITERATION"` is run for every iteration? Like 100 files per iteration or something like that, that way the limitation for long arguments is bypassed

![image](https://user-images.githubusercontent.com/11760727/72915180-cd97e180-3d37-11ea-83fa-7a77af06f641.png)

**Comment by Tapchicoma** [comment]
Good catch. While it is fixed if you just run `ktlintFormat` task via https://github.com/JLLeitschuh/ktlint-gradle/pull/278, git hook is not using it.

**Comment by manthesepeasrock** [comment]
And could the git hook create a file and pass that file instead?

**Comment by Tapchicoma** [comment]
> And could the git hook create a file and pass that file instead?

That would be general plan how to fix it, though need to look into details :slightly_smiling_face:
